packages/core/core/src/factories/useCartFactory.ts

```diff
@@ -22,29 +22,27 @@ export type UseCartFactoryParams<CART, CART_ITEM, PRODUCT, COUPON> = {
   applyCoupon: (params: {
     currentCart: CART;
     coupon: string;
-  }) => Promise<{ updatedCart: CART; updatedCoupon: COUPON }>;
+  }) => Promise<{ updatedCart: CART }>;
   removeCoupon: (params: {
```

Comment:
why you removed `coupon` property from `removeCoupon`? How will we know which one we want to remove?

Reply:
`getCouponFromCart` function will tell that.